### PR TITLE
Removed or #ifdef out unused functions

### PR DESCRIPTION
**Pull request by dpelle**
This patch removes or #ifdef out some unused functions.

More functions can be #ifdef'ed out, if there is enough interest
in doing this kind of things.

How did I find unused functions?
I used the callcatcher tool. See: https://github.com/caolanm/callcatcher

Checkout and install callcatcher:
```
$ git clone https://github.com/caolanm/callcatcher.git
$ cd callcatcher
$ sudo ./setup.py install
```

Build vim with callcatcher and run callanalyse to find unused functions:
```
$ cd vim
$ export CC="callcatcher gcc"
$ export CXX="callcatcher g++"
$ export AR="callarchive ar"

# Building different kinds of configuration of vim will
# report different unused functions. Example:
$ rm -f src/auto/config.cache
$ ./configure --with-features=huge --enable-gui=none 
$ make -j8
$ callanalyse ./src/vim
```

### CODE REVIEW

**Comment by brammool** [comment]
Dominique wrote:

> This patch removes or #ifdef out some unused functions.
> 
> More functions can be #ifdef'ed out, if there is enough interest
> in doing this kind of things.
> 
> How did I find unused functions?
> I used the callcatcher tool. See: https://github.com/caolanm/callcatcher
> 
> Checkout and install callcatcher:
> ```
> $ git clone https://github.com/caolanm/callcatcher.git
> $ cd callcatcher
> $ sudo ./setup.py install
> ```
> 
> Build vim with callcatcher and run callanalyse to find unused functions:
> ```
> $ cd vim
> $ export CC="callcatcher gcc"
> $ export CXX="callcatcher g++"
> $ export AR="callarchive ar"
> 
> # Building different kinds of configuration of vim will
> # report different unused functions. Example:
> $ ./configure --with-features=huge --enable-gui=none 
> $ make -j8
> $ callanalyse ./vim
> ```

I like this, to clean it up a bit.  So long as the #ifdefs don't get too
complicated.

We do need to add "|| defined(PROTO)" to avoid running cproto doesn't
work with all the right features enabled.


-- 
hundred-and-one symptoms of being an internet addict:
257. Your "hundred-and-one" lists include well over 101 items, since you
     automatically interpret all numbers in hexadecimal notation.
     (hex 101 = decimal 257)

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///

File: src/terminal.c [hunk 1]
```diff
@@ -2307,35 +2307,6 @@ terminal_loop(int blocking)
     return ret;
 }
 
-/*
- * Called when a job has finished.
- * This updates the title and status, but does not close the vterm, because
- * there might still be pending output in the channel.
- */
-    void
-term_job_ended(job_T *job)
```
Comment (dpelle) [inline src/terminal.c:2316]: I removed this unused function. But I wonder, perhaps it was meant to be called and there is a bug (which would be a bug prior to my change).

**Comment by brammool** [comment]
Dominique wrote:

> dpelle commented on this pull request.
> 
> > @@ -2307,35 +2307,6 @@ terminal_loop(int blocking)
>      return ret;
>  }
>  
> -/*
> - * Called when a job has finished.
> - * This updates the title and status, but does not close the vterm, because
> - * there might still be pending output in the channel.
> - */
> -    void
> -term_job_ended(job_T *job)
> 
> I removed this unused function. But I wonder, perhaps it was meant to
> be called and there is a bug (which would be a bug prior to my
> change).

I suspect this is now handled in another way, e.g. by
term_after_channel_closed().
There are a few open bugs for the terminal window, but I don't think
any of them are related to this.

-- 
hundred-and-one symptoms of being an internet addict:
266. You hear most of your jokes via e-mail instead of in person.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///

**Comment by dpelle** [comment]
@brammool  wrote:
> We do need to add "|| defined(PROTO)" 

I did that.  I also removed the WIP in the PR title.
PR is ready to be merged.

**Pull request closed without merging**